strong-config/node
# Improve 'strong-config check' to also detect deeply nested secrets

### Synopsis
fixes #43 

The first implementation of `strong-config check` was a bit naive as it only looked in the top-level of a potentially deeply-nested object.

This PR adds recursion to the search for secrets and will detect secrets at any object-level.

### How to test
1. Check out the `master` branch first
2. Add a new config file `./example/temp.yaml` with the following contents: 
   ```yaml
   application:
     url: https://myhost.com:8443/myapp/
     username: myname
     very:
       deeply:
         nestedSecret: 123
   ```
3. In the local repo's root, run `./bin/run check example/temp.yaml`
4. See that the check passes **although the file does contain a secret** 🚨🚨🚨
5. Now switch to this branch `git co fix/detect-nested-secrets` (you may need to stash your changes first and then re-apply after switching branches)
6. Run the same check again (`./bin/run check example/temp.yaml`)
7. See that this time it fails (as it should, because there is an unencrypted secret in this file) ✅
